File: 014_longest_common_prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) return new String();
        String common = strs[0];
        for(String word: strs) {
            while(common.length() > 0) {
                if(word.startsWith(common))
                    break;
                else {
                    common = common.substring(0, common.length() - 1);
                }
            }
        }
        return common;
    }
}

File: python/014_longest_common_prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs: return ""
        prefix = strs[0]
        for index, cand in enumerate(strs):
            while not strs[index].startswith(prefix):
                prefix = prefix[0:len(prefix) - 1]
                if len(prefix) <= 0: return ""
        return prefix
